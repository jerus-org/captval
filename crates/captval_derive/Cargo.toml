[package]
name = "captval_derive"
version = "0.1.0"
description = """
Derive macro for captval. Please use captval crate.
"""
authors = ["Jeremiah Russell <jrussell@jerus.ie>"]
license = "MIT OR Apache-2.0"
repository = "https://github.com/jerus-org/captval"
readme = "README.md"
documentation = "https://docs.rs/captval_derive"
categories = ["web-programming"]
keywords = ["hcaptcha", "captcha", "security", "backend", "protection"]
include = [
    "**/*.rs",
    "Cargo.toml",
    "README.md",
    "LICENSE-APACHE",
    "LICENSE-MIT",
    "CHANGES.md",
]
edition.workspace = true
rust-version.workspace = true
publish = true

[lib]
name = "captval_derive"
proc-macro = true

[dependencies]
syn.workspace = true
quote.workspace = true
proc-macro2.workspace = true
proc-macro-error2.workspace = true

[dev-dependencies]
captval = { path = "../captval" }
macrotest.workspace = true
trybuild.workspace = true

[package.metadata.docs.rs]
all-features = true
rustdoc-args = ["--cfg", "docsrs"]
