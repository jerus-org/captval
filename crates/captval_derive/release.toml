pre-release-commit-message = "chore: Release {{crate_name}} {{version}}"
tag-message = "{{tag_name}}"
tag-name = "{{prefix}}v{{version}}"
consolidate-commits = false
allow-branch = ["main"]
pre-release-replacements = [
    { file = "sonar-project.properties", search = "sonar.projectVersion=.+", replace = "sonar.projectVersion=\"{{version}}\",", exactly = 1 },
]
