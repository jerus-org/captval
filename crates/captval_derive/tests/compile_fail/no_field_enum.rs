use captval_derive::Captval;

#[derive(Captval)]
pub enum ContactEnum {
    Name,
    Token,
}

fn main() {
    println!("hello");
}
