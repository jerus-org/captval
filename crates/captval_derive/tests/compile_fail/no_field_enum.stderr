error: Must derive on a struct

         = help: This macro can only be implemented on a struct.

               #[derive(Captval)]
               struct MyStruct {
                   #[captcha]
                   captval: String,
               }

 --> tests/compile_fail/no_field_enum.rs:4:10
  |
4 | pub enum ContactEnum {
  |          ^^^^^^^^^^^
